test/libsolidity/syntaxTests/functionCalls/modifier_not_callable.sol

```diff
@@ -0,0 +1,7 @@
+contract C {
+    uint a = m(1000);
+
+    modifier m(uint) { _; }
+}
+// ----
+// TypeError 5704: (26-33): This object is not callable.
```

Comment:
Calling it an "object" might not be entirely correct either, but I'm not sure what to call it :) Do we consider all these things objects?

I could define a helper to give a string representation to `Type::Category` and use it here so it would specifically say "modifier", "module", "magic variable", etc., but I thought it would be an overkill.

"Object" is at least less wrong than "type".

Reply:
The `Type::Category` approach would be ideal, and not overkill in my opinion. You are right that calling a `modifier` an `object` is less wrong than calling it a `type`, but it's still a bit iffy. I would rather this be properly fixed as you suggested, but I'm not sure whether merging this PR as an incremental improvement is good or not.

Reply:
Well, ok then. I'll add it.

I meant overkill just for this message, but overall, I think we should have such a function already. Would have been useful many times at least for debugging.

Reply:
Done.